modules/phase_field/src/base/PhaseFieldApp.C

```diff
@@ -161,6 +162,8 @@ PhaseFieldApp::registerApps()
   registerApp(PhaseFieldApp);
 }
 
+// External entry point for dynamic object registration
+extern "C" void PhaseFieldApp__registerObjects(Factory & factory) { PhaseFieldApp::registerObjects(factory); }
```

Comment:
Quite frankly that will be easier. Our registration macro makes it difficult to know what type of object each name belongs too, furthermore the user might simply be using the wrong one which is perfectly valid in MOOSE! We don't check to make sure that things match up until runtime. I'm happy to go that route, seems even better and as long as we support a vector of those objects (or just a space delimited string) we should be flexible enough for future needs.